integration/simulation/simulation_in_mem_test.go

```diff
@@ -24,11 +36,78 @@ func TestInMemoryMonteCarloSimulation(t *testing.T) {
+
+	tmpClient := buildhelper.NewEthAPI("127.0.0.1", 7545)
+	err := tmpClient.Connect()
+	if err != nil {
+		panic(err)
+	}
+
+	deployContractTx := types.LegacyTx{
+		Nonce:    0, // relies on a clean env
+		GasPrice: big.NewInt(2000000000),
+		Gas:      1025_000_000,
+		Data:     common.Hex2Bytes(contractByte),
+	}
+
+	signedTx, err := types.SignNewTx(buildhelper.Addr1PK(), types.NewEIP155Signer(big.NewInt(1337)), &deployContractTx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = tmpClient.ApiClient.SendTransaction(context.Background(), signedTx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var receipt *types.Receipt
+	for start := time.Now(); time.Since(start) < 10*time.Second; time.Sleep(time.Second) {
+		receipt, err = tmpClient.ApiClient.TransactionReceipt(context.Background(), signedTx.Hash())
+		if err == ethereum.NotFound {
```

Comment:
@otherview I personally prefer the following (it feels unnatural for the "standard" path to be to break the loop), but your call:

		if err == nil {
			break
		}
		if err != ethereum.NotFound {
			t.Fatal(err)
		}
		t.Logf("Contract deploy tx has not been mined into a block after %s...", time.Since(start))

Reply:
Yeah good point 👍